server/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -214,6 +216,17 @@ void restoreShard(Store store, SnapshotId snapshotId, IndexId indexId, ShardId s
      */
     IndexShardSnapshotStatus getShardSnapshotStatus(SnapshotId snapshotId, IndexId indexId, ShardId shardId);
 
+    /**
+     * Check if this instances {@link Settings} can be changed to the provided updated settings without recreating the repository.
+     *
+     * @param updatedSettings new repository settings
+     * @param ignoredSettings setting names to ignore even if changed
+     * @return true if the repository can be updated in place
+     */
+    default boolean canUpdateInPlace(Settings updatedSettings, Set<String> ignoredSettings) {
```

Comment:
It seems I should have documented this better :) It's so we can do dynamic setting updates on nested repositories. E.g. what this is intended to be used for:

Encrypted repo wraps some other blob store repository. We want to update a setting `some.key.indentifier` dynamically there but want this to not cause say the S3 repository to fail an in-place update. An alternative to this would be to say we specifically check all supported setting values for changes and ignore noop change to settings that the repo ignores anyway, but with the diversity of settings across repos that we have that turned out to be a very large change and a lot of work (retry backoffs and whatnot are all different across implementations).
-> I went with this kind of inverted approach for now to get this done without too much noise though long-term doing it the other way around seems nicer obviously (IMO).